lsp/src/Language/LSP/Server/Processing.hs

```diff
@@ -330,7 +336,9 @@ handle :: (m ~ LspM config) => LogAction m (WithSeverity LspProcessingLog) -> SC
 handle logger m msg =
   case m of
     SMethod_WorkspaceDidChangeWorkspaceFolders -> handle' logger (Just updateWorkspaceFolders) m msg
-    SMethod_WorkspaceDidChangeConfiguration    -> handle' logger (Just $ handleConfigChange logger) m msg
+    SMethod_WorkspaceDidChangeConfiguration    -> handle' logger (Just $ handleDidChangeConfiguration logger) m msg
+    -- See Note [LSP configuration]
+    SMethod_Initialized                        -> handle' logger (Just $ \_ -> requestConfigUpdate (cmap (fmap LspCore) logger)) m msg
```

Comment:
We are requesting config update for every time initialized, considering downstream users, should we mark this as optional?

Reply:
`initialized` gets sent only once, which is when client receives our response to `initialize`. So this is a one-off thing.